client/src/js/todos/dragAndDrop.js

```diff
@@ -0,0 +1,154 @@
+    this.clonedDragCard.style.left = mouseX + 'px';
+    this.clonedDragCard.style.top = mouseY + 'px';
+
+    const cloneCardPosData = this.clonedDragCard.getBoundingClientRect();
+    const columnPosData = this.column.getBoundingClientRect();
+
+    const draggedX = cloneCardPosData.x;
+    const draggedY = cloneCardPosData.y;
+
+    if (columnPosData.x - cloneCardPosData.x > cloneCardPosData.width / 2) {
+      $(
+        '.column-list',
+        this.column.parentElement.previousElementSibling
+      ).appendChild(this.dragCard);
+    } else if (
+      columnPosData.x + columnPosData.width - cloneCardPosData.x <
+      cloneCardPosData.width / 2
+    ) {
+      $(
+        '.column-list',
+        this.column.parentElement.nextElementSibling
+      ).appendChild(this.dragCard);
+    }
+
+    this.cardPositionInfo.forEach((el) => {
+      const prevCard = el.x - draggedX > 0 ? cloneCardPosData : el;
+      const curCard = el.x - draggedX > 0 ? el : cloneCardPosData;
+
+      const higherCard = el.y - draggedY > 0 ? cloneCardPosData : el;
+      const lowerCard = el.y - draggedY > 0 ? el : cloneCardPosData;
+
+      if (
+        this.dragCard.getBoundingClientRect().x === el.x &&
+        !this.moveCardState
+      ) {
+        if (prevCard.x + prevCard.width - curCard.x > prevCard.width / 2) {
+          if (
+            cloneCardPosData.y < el.y &&
+            higherCard.y + higherCard.height - lowerCard.y >
+              lowerCard.height / 2
+          ) {
+            el.positionedCard.insertAdjacentElement('afterend', this.dragCard);
+          }
+
+          if (
+            cloneCardPosData.y > el.y &&
+            higherCard.y + higherCard.height - lowerCard.y >
+              higherCard.height / 2
+          ) {
+            el.positionedCard.insertAdjacentElement(
+              'beforebegin',
+              this.dragCard
+            );
+          }
+        }
+      } else {
+        this.moveCardState = true;
+
+        if (prevCard.x + prevCard.width - curCard.x > prevCard.width / 2) {
+          if (
+            cloneCardPosData.y < el.y &&
+            higherCard.y + higherCard.height - lowerCard.y >
+              lowerCard.height / 2
+          ) {
+            el.positionedCard.insertAdjacentElement(
+              'beforebegin',
+              this.dragCard
+            );
+          }
+
+          if (
+            cloneCardPosData.y > el.y &&
+            higherCard.y + higherCard.height - lowerCard.y >
+              higherCard.height / 2
+          ) {
+            el.positionedCard.insertAdjacentElement('afterend', this.dragCard);
+          }
+        }
+      }
+    });
+  };
+
+  setCardsPosition() {
+    for (const li of $$('.list_item')) {
+      this.cardPositionInfo.push({
+        x: li.getBoundingClientRect().x,
+        y: li.getBoundingClientRect().y,
+        width: li.getBoundingClientRect().width,
+        height: li.getBoundingClientRect().height,
+        positionedCard: li,
```

Comment:
distructuring 사용하여 축약형으로 표현 가능하겠네요~
```suggestion
      const { x, y, width, height } = li.getBoundingClientRect();
      this.cardPositionInfo.push({
        x,
        y,
        width,
        height,
        positionedCard: li,
```